api/users/users-model.js

```diff
@@ -1,29 +1,38 @@
+const db = require(`../../data/db-config`)
+
 /**
   resolves to an ARRAY with all users, each user having { user_id, username }
  */
 function find() {
-
+  return db('users').select(`user_id`, `username`)
 }
 
 /**
   resolves to an ARRAY with all users that match the filter condition
  */
 function findBy(filter) {
-
+  return db(`users`).where(filter).first()
 }
 
 /**
   resolves to the user { user_id, username } with the given user_id
  */
 function findById(user_id) {
-
+  return db(`users`).where(`user_id`, user_id).first()
```

Comment:
Another way to write this is if the parameter is the same name as the table column you could do `.where({user_id})` as well